Scripts/Python/grsnWallPython.py

```diff
@@ -329,53 +331,161 @@ def OnServerInitComplete(self):
+                        if currentgender == kFemaleClothingGroup:
+                            clothing = FemaleSuit
+                        else:
+                            clothing = MaleSuit
+                        if not wornItem:
+                            wornItem = avatar.avatar.getAvatarClothingList()
+                            DefaultColor1 = {item[1]: avatar.avatar.getTintClothingItem(item[0],1) for item in wornItem}
+                            DefaultColor2 = {item[1]: avatar.avatar.getTintClothingItem(item[0],2) for item in wornItem}
+                        for index, item in enumerate(clothing):
+                            if item == None:
+                                continue
+                            avatar.avatar.netForce(1)
+                            avatar.avatar.wearClothingItem(item, 0)
+                            avatar.avatar.tintClothingItem(item, DefaultColor1[index], 0)
+                            avatar.avatar.tintClothingItemLayer(item, DefaultColor2[index], 2, 1)
+                        PtSendKIMessage(kDisableEntireYeeshaBook, 0)
+                        avatar.physics.warpObj(NorthTeamWarpPt.value.getKey())
+                        ageSDL["sWallPlayer"] = (-1,)
+                        if (ageSDL["sState"][0] == kEntry):
+                            self.ChangeGameState(kNorth, kGameInProgress)
+                            self.ChangeGameState(kSouth, kGameInProgress)
+                    SouthTubeOpen.run(self.key)
+                    return
+            elif (id == NorthQuitBehavior.id):
+                if (event[0] == kMultiStageEvent and event[1] == 0 and event[2] == kEnterStage and PtFindAvatar(events) == avatar):
+                    #Touched the Crystal
+                    PtDebugPrint("Trigger North Crystal")
+                    PtAtTimeCallback(self.key, 3.0, kLinkToNorthNexus)
+                    return
+            elif (id == SouthQuitBehavior.id):
+                if (event[0] == kMultiStageEvent and event[1] == 0 and event[2] == kEnterStage and PtFindAvatar(events) == avatar):
+                    #Touched the Crystal
+                    PtDebugPrint("Trigger South Crystal")
+                    PtAtTimeCallback(self.key, 3.0, kLinkToSouthNexus)
+                    return
+            ### Blocker ###
+            elif (id == NorthPanelClick.id or id == SouthPanelClick.id):
+                if (event[0] == kPickedEvent and event[1] == 1):
+                    pickedBlockerObj = event[3]
+                    team = kNorth
+                    try:
+                        index = NorthPanel.value.index(pickedBlockerObj)
+                    except:
+                        try:
+                            index = SouthPanel.value.index(pickedBlockerObj)
+                            team = kSouth
+                        except:
+                            PtDebugPrint("grsnWallPython::OnNotify: Blocker not found on either panel")
+                            continue
```

Comment:
This seems a bit sus to me. I feel like we should know for certain which panel we're working with, not trying one then trying the other for funsies. 😕

Reply:
I wish, it returns a scene object.
I could probably get the name of the object and parse north or south from that as every panel object starts with N or S

Reply:
could also seperate the check into north and south instead of combining them, would make the south panel one less index to check

Reply:
Yeah, it would probably be a good idea to set the team by doing something like:
```python
if id == NorthPanelClick.id:
    team = kNorth
    blockerObjs = NorthPanel.value
else:
    team = kSouth
    blockerObjs = SouthPanel.value
index = blockerObjs.index(pickedBlockerObjs)
```

Reply:
Do you think it would be better to set the north and south panels to a constant on age initialization instead of setting up a dictionary everytime?